lib/src/upZenohClient.cpp

```diff
@@ -51,4 +75,8 @@ std::shared_ptr<UpZenohClient> UpZenohClient::instance(void) noexcept {
     }
 }
 
+void UpZenohClient::purge() noexcept
+{
+    w_handle.reset();
+}
```

Comment:
Why is this even necessary? `w_handle` is a weak pointer - the intent is that it _automatically_ gets reset when all shared pointers to that instance are lost. Calling reset on this here _doesn't_ clean up any existing instances.